test/parallel/test-net-server-max-connections-close-makes-more-available.js

```diff
@@ -0,0 +1,85 @@
+'use strict';
+var common = require('../common');
+var assert = require('assert');
+
+var net = require('net');
+
+// Sets the server's maxConnections property to 1. 
+// Open 2 connections (connection 0 and connection 1). 
+// Connection 0  should be accepted.
+// Connection 1 should be rejected.
+// Closes connection 0.
+// Open 2 more connections (connection 2 and 3).
+// Connection 2 should be accepted.
+// Connection 3 should be rejected.
+
+var connections = [];
+var received = [];
+var sent = [];
+
+var createConnection = function(index) {
+  console.error('creating connection ' + index);
+
+  return new Promise(function(resolve, reject) {
+    var connection = net.createConnection(common.PORT, function() {
+      var msg = '' + index;
+      console.error('sending message: ' + msg);
+      this.write(msg);
+      sent.push(msg);
+    });
+
+    connection.on('data', function(e) {
+      console.error('connection ' + index + ' received response');
+      resolve();
+    });
+
+    connection.on('end', function() {
+      console.error('ending ' + index);
+      resolve();
```

Comment:
According to [2.3.3.3.3 in Promises A+ spec](https://promisesaplus.com/#point-59), if `resolve()` is called a second time, the second call is has no effect.